testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/messaging/CoreBridgeCallTimeoutTestCase.java

```diff
@@ -0,0 +1,242 @@
+import static org.jboss.as.controller.client.helpers.ClientConstants.OP;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * This test verifies that setting a call timeout on a core bridge takes an effect.
+ *
+ * Scenario:
+ *
+ * <ol>
+ * <li>Create a bridge without a call-timeout.</li>
+ * <li>Verify that messages are transferred.</li>
+ * <li>Introduce a delay to bridge calls via a byteman rule and recreate bridge with call-timeout value of 500 ms.</li>
+ * <li>Verify that messages are not transferred.</li>
+ * </ol>
+ *
+ * @author Tomas Hofman
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class CoreBridgeCallTimeoutTestCase {
+
+    private static final Logger logger = Logger.getLogger(CoreBridgeCallTimeoutTestCase.class);
+
+    private static final String CONTAINER_QUALIFIER = "default-full-jbossas-byteman";
+    private static final String EXPORTED_PREFIX = "java:jboss/exported/";
+    private static final String REMOTE_CONNECTION_FACTORY_LOOKUP = "jms/RemoteConnectionFactory";
+    private static final String SOURCE_QUEUE_NAME = "SourceQueue";
+    private static final String SOURCE_QUEUE_LOOKUP = "jms/SourceQueue";
+    private static final String TARGET_QUEUE_NAME = "TargetQueue";
+    private static final String TARGET_QUEUE_LOOKUP = "jms/TargetQueue";
+    private static final String BRIDGE_NAME = "CoreBridge";
+    private static final String MESSAGE_TEXT = "Hello world!";
+
+    private static final Submit BYTEMAN = new Submit(
+            System.getProperty("byteman.server.ipaddress", Submit.DEFAULT_ADDRESS),
+            Integer.getInteger("byteman.server.port", Submit.DEFAULT_PORT));
+
+    @ArquillianResource
+    protected static ContainerController container;
+
+    private JMSOperations jmsOperations;
+    private ManagementClient managementClient;
+
+    @Before
+    public void setUp() throws Exception {
+        if (!container.isStarted(CONTAINER_QUALIFIER)) {
+            container.start(CONTAINER_QUALIFIER);
+        }
+
+        managementClient = createManagementClient();
+        jmsOperations = JMSOperationsProvider.getInstance(managementClient.getControllerClient());
+
+        // create source and target queues and a bridge
+        jmsOperations.createJmsQueue(SOURCE_QUEUE_NAME, EXPORTED_PREFIX + SOURCE_QUEUE_LOOKUP);
+        removeMessagesFromQueue(SOURCE_QUEUE_NAME);
+        jmsOperations.createJmsQueue(TARGET_QUEUE_NAME, EXPORTED_PREFIX + TARGET_QUEUE_LOOKUP);
+        removeMessagesFromQueue(TARGET_QUEUE_NAME);
+    }
+
+    @After
+    public void tearDown() {
+        // clean up
+        jmsOperations.removeJmsQueue(SOURCE_QUEUE_NAME);
+        jmsOperations.removeJmsQueue(TARGET_QUEUE_NAME);
+        jmsOperations.removeCoreBridge(BRIDGE_NAME);
+        container.stop(CONTAINER_QUALIFIER);
```

Comment:
Why we need to stop the container after each test? What about `s/@After/@AfterClass/` and the same for setUp method? Btw, why we actually need to have this test in manual mode?

Reply:
Ah, manualmode is needed because of byteman arquillian container. But still I hope that we could remove one reload by `@After` -> `@AfterClass` and related changes

Reply:
I not happy with the manualmode module either. The reason I put it there is that Emmanuel already has another PR with byteman integration that will go into the manualmode module.

I don't think it's desirable to have byteman in more modules than is necessary. But if you think otherwise I can move it.

I will look into the restarts, agree on that.

Reply:
@marekkopecky I implemented it via test method ordering, PR is updated.